cli/priority.rs

```diff
@@ -0,0 +1,159 @@
+      errno(),
+    ) {
+      (-1, Errno(0)) => Ok(PRIORITY_HIGH),
+      (-1, _) => Err(OpError::from(std::io::Error::last_os_error())),
+      (priority, _) => Ok(priority),
+    }
+  }
+}
+
+#[cfg(unix)]
+pub fn set_priority(pid: u32, priority: i32) -> Result<(), OpError> {
+  unsafe {
+    match libc::setpriority(PRIO_PROCESS as priority_t, pid as id_t, priority) {
+      -1 => Err(OpError::from(std::io::Error::last_os_error())),
+      _ => Ok(()),
+    }
+  }
+}
+
+#[cfg(windows)]
+pub fn get_priority(pid: u32) -> Result<i32, OpError> {
+  unsafe {
+    let handle = if pid == 0 {
+      GetCurrentProcess()
+    } else {
+      OpenProcess(PROCESS_QUERY_LIMITED_INFORMATION, FALSE, pid as DWORD)
+    };
+    if handle == NULL {
+      Err(OpError::from(std::io::Error::last_os_error()))
+    } else {
+      let result = match GetPriorityClass(handle) {
+        0 => Err(OpError::from(std::io::Error::last_os_error())),
+        REALTIME_PRIORITY_CLASS => Ok(PRIORITY_HIGHEST),
+        HIGH_PRIORITY_CLASS => Ok(PRIORITY_HIGH),
+        ABOVE_NORMAL_PRIORITY_CLASS => Ok(PRIORITY_ABOVE_NORMAL),
+        NORMAL_PRIORITY_CLASS => Ok(PRIORITY_NORMAL),
+        BELOW_NORMAL_PRIORITY_CLASS => Ok(PRIORITY_BELOW_NORMAL),
+        IDLE_PRIORITY_CLASS => Ok(PRIORITY_LOW),
+        _ => Ok(PRIORITY_LOW),
+      };
+      CloseHandle(handle);
+      result
+    }
+  }
+}
+
+#[cfg(windows)]
+pub fn set_priority(pid: u32, priority: i32) -> Result<(), OpError> {
+  unsafe {
+    let handle = if pid == 0 {
+      GetCurrentProcess()
+    } else {
+      OpenProcess(PROCESS_QUERY_LIMITED_INFORMATION, FALSE, pid as DWORD)
+    };
+    if handle == NULL {
+      Err(OpError::from(std::io::Error::last_os_error()))
+    } else {
+      let prio_class = match priority {
+        p if p <= PRIORITY_HIGHEST => REALTIME_PRIORITY_CLASS,
+        p if PRIORITY_HIGHEST < p && p <= PRIORITY_HIGH => HIGH_PRIORITY_CLASS,
+        p if PRIORITY_HIGH < p && p <= PRIORITY_ABOVE_NORMAL => {
+          ABOVE_NORMAL_PRIORITY_CLASS
+        }
+        p if PRIORITY_ABOVE_NORMAL < p && p <= PRIORITY_NORMAL => {
+          NORMAL_PRIORITY_CLASS
+        }
+        p if PRIORITY_NORMAL < p && p <= PRIORITY_BELOW_NORMAL => {
+          BELOW_NORMAL_PRIORITY_CLASS
+        }
+        _ => IDLE_PRIORITY_CLASS,
+      };
+      let result = match SetPriorityClass(handle, prio_class) {
+        FALSE => Err(OpError::from(std::io::Error::last_os_error())),
+        _ => Ok(()),
+      };
+      CloseHandle(handle);
+      result
+    }
+  }
+}
+
+#[cfg(test)]
+mod tests {
+  use super::*;
+
+  #[test]
+  fn test_get_current_process_priority() {
+    get_priority(0).expect("Should get priority");
+  }
+
+  #[cfg(unix)]
+  #[test]
+  fn test_set_current_process_high_priority_should_fail() {
+    assert!(set_priority(0, PRIORITY_HIGH).is_err());
+  }
+
+  /// this test makes multiple tests at once
+  /// because we need to set them in order and rust
+  /// does not garanty test order execution
```

Comment:
minor typo.  'guarantee'

Reply:
yes, i'm not a native speaker, so i tend to do some of these.

Reply:
No worries.  I am a native speaker, and I still made a typo mistake the other day.  I've now installed the 'Code Spell Checker' extension in Visual Studio Code which helps me spell correctly now.